widgetastic_manageiq.py

```diff
@@ -878,3 +879,280 @@ def last_page(self):
+        self.minus_button.click()
+
+    def increase(self):
+        self.plus_button.click()
+
+    def set_value(self, value):
+        value = int(value)
+        if value < 1:
+            raise ValueError('The value cannot be less than 1')
+
+        steps = value - self.read()
+        if steps == 0:
+            return False
+        elif steps > 0:
+            operation = self.increase
+        else:
+            operation = self.decrease
+
+        steps = abs(steps)
+        for step in range(steps):
+            operation()
+        return True
+
+    def fill(self, value):
+        return self.set_value(value)
+
+
+class RadioButton(Button):
+    """CFME Radio Button Control
+       it should be mainly used in RadioGroup control
+    """
+    ROOT = ParametrizedLocator('.//*[self::input and @type="radio" and '
+                               'parent::label[normalize-space(.)={name|quote}]]')
+
+    def __init__(self, parent, name, logger=None):
+        Button.__init__(self, parent=parent, logger=logger)
+        self._name = name
+
+    @property
+    def is_selected(self):
+        if 'ng-valid-parse' in self.browser.classes(self):
+            return True
+        return False
+
+    @property
+    def name(self):
+        return self._name
+
+
+class RadioGroup(Widget):
+    """ CFME Radio Group Control
+
+        .. code-block:: python
+        radio_group = RadioGroup(locator='//span[contains(@class, "timeline-option")]')
+        radio_group.select(radio_group.button_names()[-1])
+    """
+    def __init__(self, parent, locator, logger=None):
+        Widget.__init__(self, parent=parent, logger=logger)
+        self._locator = locator
+
+    def __locator__(self):
+        return self._locator
+
+    @property
+    def button_names(self):
+        buttons = self.browser.elements('//label[input[@type="radio"]]')
+        return [self.browser.text(btn) for btn in buttons]
+
+    @property
+    def buttons(self):
+        return [RadioButton(self, name=name) for name in self.button_names]
+
+    @property
+    def selected(self):
+        return [btn for btn in self.buttons if btn.is_selected][-1].name
+
+    def select(self, name):
+        button = [btn for btn in self.buttons if btn.name == name][-1]
+        if not button.is_selected:
+            button.click()
+            return True
+        return False
+
+    def read(self):
+        return self.selected
+
+    def fill(self, name):
+        return self.select(name)
+
+
+class BreadCrumb(Widget):
+    """ CFME BreadCrumb navigation control
+
+        .. code-block:: python
+        breadcrumb = BreadCrumb()
+        breadcrumb.click_location(breadcrumb.locations[0])
+    """
+    ROOT = '//ol[@class="breadcrumb"]'
+    ELEMENTS = './/li'
+
+    def __init__(self, parent, locator=None, logger=None):
+        Widget.__init__(self, parent=parent, logger=logger)
+        self._locator = locator or self.ROOT
+
+    def __locator__(self):
+        return self._locator
```

Comment:
You don't need to have `__locator__` if you have ROOT, that is then automatically considered as the root locator.